conjure-java-core/src/integrationInput/java/com/palantir/product/StrictOneCollectionField.java

```diff
@@ -0,0 +1,150 @@
+                && this.memoizedHashCode != other.memoizedHashCode) {
+            return false;
+        }
+        return this.myList.equals(other.myList);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = memoizedHashCode;
+        if (result == 0) {
+            result = this.myList.hashCode();
+            memoizedHashCode = result;
+        }
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "StrictOneCollectionField{myList: " + myList + '}';
+    }
+
+    public static StrictOneCollectionField of(List<String> myList) {
+        return builder().myList(myList).build();
+    }
+
+    private static void validateFields(List<String> myList) {
+        List<String> missingFields = null;
+        missingFields = addFieldIfMissing(missingFields, myList, "myList");
+        if (missingFields != null) {
+            throw new SafeIllegalArgumentException(
+                    "Some required fields have not been set", SafeArg.of("missingFields", missingFields));
+        }
+    }
+
+    private static List<String> addFieldIfMissing(List<String> prev, Object fieldValue, String fieldName) {
+        List<String> missingFields = prev;
+        if (fieldValue == null) {
+            if (missingFields == null) {
+                missingFields = new ArrayList<>(1);
+            }
+            missingFields.add(fieldName);
+        }
+        return missingFields;
+    }
+
+    public static MyListStageBuilder builder() {
+        return new DefaultBuilder();
+    }
+
+    public interface MyListStageBuilder {
+        Completed_StageBuilder myList(@Nonnull List<String> myList);
+
+        Builder from(StrictOneCollectionField other);
+    }
+
+    public interface Completed_StageBuilder {
+        StrictOneCollectionField build();
+    }
+
+    public interface Builder extends MyListStageBuilder, Completed_StageBuilder {
+        @Override
+        Builder myList(@Nonnull List<String> myList);
+
+        @Override
+        Builder from(StrictOneCollectionField other);
+
+        @Override
+        StrictOneCollectionField build();
+    }
+
+    @Generated("com.palantir.conjure.java.types.BeanBuilderGenerator")
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    static final class DefaultBuilder implements Builder {
+        boolean _buildInvoked;
+
+        private List<String> myList = new ArrayList<>();
+
+        private DefaultBuilder() {}
+
+        @Override
+        public Builder from(StrictOneCollectionField other) {
+            checkNotBuilt();
+            myList(other.getMyList());
+            return this;
+        }
+
+        @Override
+        @JsonSetter(value = "myList", nulls = Nulls.SKIP)
+        public Builder myList(@Nonnull List<String> myList) {
+            checkNotBuilt();
+            this.myList = ConjureCollections.newArrayList(Preconditions.checkNotNull(myList, "myList cannot be null"));
+            return this;
+        }
+
```

Comment:
I've removed the auxiliary methods for lists (e.g. `addAllX`). I think it makes sense to enforce users that users not mutate the fields after construction when using strict staged builders. What do you think?